File: JavaAlgorithm/src/main/java/algo/tzashinorpu/SpecialSubject/Array_List/DoublePointer/Array/removeElement_27.java
package algo.tzashinorpu.SpecialSubject.Array_List.DoublePointer.Array;

public class removeElement_27 {
    public int removeElement(int[] nums, int val) {
        int p = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != val) nums[p++] = nums[i];
        }
        return p;
    }
}

File: Clion/ZeroTrac/array_list/removeElement_27.cpp
//
// Created by TzashiNorpu on 10/18/2022.
//
#include "array_list.h"

using namespace Array_List;

int Solution::removeElement(vector<int> &nums, int val) {
  int fast = 0, slow = 0;
  while (fast < nums.size()) {
    if (nums[fast] != val) nums[slow] = nums[fast], slow++;
    fast++;
  }
  return slow;
}